### ISSUE TITLE
Proper documentation

### ISSUE THREAD

**Issue by janwirth** [SE7ENSKY/jade2php#10]
- [ ] API (phpRuntime options)
- [x] Tests
- [x] Contributing

**Comment by janwirth** [comment]
I suppose using <a href='https://readthedocs.org'> this platform </a> for documentation.
It features GitHub integration.

**Comment by ivankravchenko** [comment]
I'm not sure how rtfd platform can bring a value for now.
Describe in more detail this and what is essential in 'Contributing' section.

**Comment by janwirth** [comment]
I guess you are right about the rtfd platform, that would be overkill as this is mostly jade. Yet still we have to document the phpRuntime options, and how functions are called inside tags.

Contributing is mainly to keep a consistent quality standard throughout this library:
https://github.com/jenius/accord/blob/master/contributing.md

**Comment by ivankravchenko** [comment]
Of course, PHP runtime code, extractor code, detailed options and usage should be well documented.
Thanks for pointing on well-formed 'Contributing' section.

**Comment by OnkelTem** [comment]
Meanwhie, would you explain what exactly jade2php generates?

Rendering this simple template:

``` jade
doctype html
html(lang!="$language")
  head
    !=$head
```

using: `./bin/jade2php --pretty --omit-php-runtim --omit-php-extractor html.jade`
results to:

``` php
<?php error_reporting(E_ALL & ~E_NOTICE);function attr($name, $value = true, $escaped = true) {if (!empty($value)) {echo " $name=\"$value\""
;}}function attrs() {$args = func_get_args();$attrs = array();foreach ($args as $arg) {foreach ($arg as $key => $value) {if ($key == 'class'
) {if (!isset($attrs[$key])) $attrs[$key] = array();$attrs[$key] = array_merge($attrs[$key], is_array($value) ? $value : explode(' ', $value
));} else {$attrs[$key] = $value;}}}foreach ($attrs as $key => $value) {if ($key == 'class') {attr_class($value);} else {attr($key, $value);
}}}function attr_class() {$classes = array();$args = func_get_args();foreach ($args as $arg) {if (empty($arg) || is_array($arg) && count($ar
g) == 0) continue;$classes = array_merge($classes, is_array($arg) ? $arg : array($arg));}$classes = array_filter($classes);if (count($classe
s) > 0) attr('class', join(' ', $classes));}function add() {$result = '';$args = func_get_args();$concat = false;foreach ($args as $arg) {if
 ($concat || is_string($arg)) {$concat = true;$result .= $arg;} elseif (is_numeric($arg)) {if ($result === '') $result = 0;$result += $arg;}
}return $result;}?><!DOCTYPE html><html lang="$language"><head><?= $$head ?></head></html>
```

![](https://sc-cdn.scaleengine.net/i/4ea31b2eda16ecb5679e6d517a132d66.png)

_UPDATE_

Fixed "--omit-php-runtim" in command line, now the output is much nicer:

``` php
<!DOCTYPE html><html lang="$language"><head><?= $$head ?></head></html>
```

but still I don't get how to make it look like:

``` php
<!DOCTYPE html>
  <html lang="<?php echo $language?>">
    <head>
      <?php echo $head ?>
    </head>
  </html>
```

i.e.:
1. with newlines and proper indentation
2. with `<?php` tags and `echo` (or `print`) instead of  `<?=`
3. with resolved attributes

Thanks in advance!

**Comment by ivankravchenko** [comment]
1, 2. this is `--pretty`-related, #14.
3. What do you mean at "resolved attributes"? Could you mind to describe at code level with input and output?

**Comment by janwirth** [comment]
@OnkelTem
you should write jade in the regular syntax, e.g. your <code>$variable</code> must not have a sigil (the dollar sign). Just go with <code>variable</code>. Jade2Php will then automatically wrap your variable inside the <code> htmlspecialchars() </code> function.

**Comment by OnkelTem** [comment]
Guys, I'm trying to figure out the difference between jade-php and jade2php.
Form the source code point of view, while jade-php is 30-lines jade's Compile-callback implementation, jade2php is huge. Would you explain jade2php purpose and idea?

@FranzSkuffka
I see, ok :)

**Comment by ivankravchenko** [comment]
`jade2php` strives to be as compatible with original jade code as possible, while `jade-php` don't.
I made and used `jade2php` to write interoperable code (can be used with original Jade and with PHP).
HTML output is also byte-to-byte same as with usage of original Jade preprocessor.

While writing `jade-php`-ish Jade code, you can't use it without `jade-php`.

**Comment by ivankravchenko** [comment]
One more important part – `jade2php` is extensively unit-tested.
There also some edge cases and work to do, but I could say that `jade2php` is ready to be used (and open for contributions).

**Comment by kevindice** [comment]
I might be interested in helping document this.

**Comment by ivankravchenko** [comment]
@kevindice, you're welcome. Let do this via usual Fork → Pull Request?

**Comment by janwirth** [comment]
@ivankravchenko I just created http://pug2php.netlify.com/ which is a small demo app. What do you think?

@kevindice Are you still interested in creating a documentation?

Repo: https://github.com/FranzSkuffka/pug2php-www
@ivankravchenko do you want me to transfer this to se7ensky?

**Comment by dsifford** [comment]
So I'm interested in the gulp plugin.. Unfortunately there is no documentation whatsoever for it (not even a readme.md).

Am I able to pass args using gulp?

**Comment by dsifford** [comment]
Scratch my last... Read through the source file and it looks like the options should be given in camelCase rather than kebab-case... 

Docs would still be handy though.
